internal/sbi/producer/ue_context_management.go

```diff
@@ -245,23 +261,13 @@ func RegistrationAmf3gppAccessProcedure(registerRequest models.Amf3GppAccessRegi
 		}
 	}()
 
-	// TS 23.502 4.2.2.2.2 14d: UDM initiate a Nudm_UECM_DeregistrationNotification to the old AMF
-	// corresponding to the same (e.g. 3GPP) access, if one exists
-	if oldAmf3GppAccessRegContext != nil {
-		deregistData := models.DeregistrationData{
-			DeregReason: models.DeregistrationReason_SUBSCRIPTION_WITHDRAWN,
-			AccessType:  models.AccessType__3_GPP_ACCESS,
-		}
-		callback.SendOnDeregistrationNotification(ueID, oldAmf3GppAccessRegContext.DeregCallbackUri,
-			deregistData) // Deregistration Notify Triggered
-
-		return nil, nil, nil
-	} else {
+	if oldAmf3GppAccessRegContext == nil {
```

Comment:
I think this condition is unnecessary because the UDM sent the dereg notification to the old AMF.

Reply:
When the UE is initially registered with the UDM through the AMF, it triggers the execution of this function block.

Reply:
no matter dereg sent or not, it should always `return header, &registerRequest, nil`

Reply:
try this:
```
	// TS 23.502 4.2.2.2.2 14d: UDM initiate a Nudm_UECM_DeregistrationNotification to the old AMF
	// corresponding to the same (e.g. 3GPP) access, if one exists
	if oldAmf3GppAccessRegContext != nil {
		deregistData := models.DeregistrationData{
			DeregReason: models.DeregistrationReason_SUBSCRIPTION_WITHDRAWN,
			AccessType:  models.AccessType__3_GPP_ACCESS,
		}

		logger.UecmLog.Infof("Send DeregNotif to old AMF GUAMI=%v", oldAmf3GppAccessRegContext.Guami)
		go func() {
			pd := callback.SendOnDeregistrationNotification(ueID,
				oldAmf3GppAccessRegContext.DeregCallbackUri,
				deregistData) // Deregistration Notify Triggered
			if pd != nil {
				logger.UecmLog.Errorf("RegistrationAmf3gppAccess: send DeregNotif fail %v", pd)
			}
		}()
		return nil, &registerRequest, nil
	} else {
		header = make(http.Header)
		udmUe, _ := udm_context.Getself().UdmUeFindBySupi(ueID)
		header.Set("Location", udmUe.GetLocationURI(udm_context.LocationUriAmf3GppAccessRegistration))
		return header, &registerRequest, nil
	}
```

Reply:
see https://github.com/free5gc/udm/pull/25#discussion_r1323787713